doc/gwsettings.md

```diff
@@ -29,7 +29,7 @@ then, starting at `size`, the following algorithm is employed:
 ```c
 uint32_t hcs32(const char* buf, size_t len)
 {
-	uint32_t checksum = 0;
+	uint32_t checksum = len + 8;
```

Comment:
This is only correct if we start the checksum calculation at `.data`, and assume the header format presented above, with the `.checksum` member set to zero. The checksum algorithm presented here is agnostic of the underlying buffer format, and thus more descriptive.

There's [a comment in the source code](https://github.com/jclehner/bcm2-utils/blob/fc751c14f091ee85fbfafe3f4802477e40e9403f/gwsettings.cc#L579) explaining the different approaches.

Reply:
When trying to recreate this code in Python, it fails to give the correct checksum, whereas adding the length fixes it.  Am I giving it the wrong parameters?

Reply:
I'm guessing you're only passing it the "data" part, whereas you should include the two header fields, with `.checksum` set to `0`. If you want  to exclude the header, then setting the initial checksum to `length + 8` is correct.

To quote from `gwsettings.md`:

| Offset | Type         | Name       |
|-------:|--------------|------------|
|  `0`   | `u32`        | `size`     |
|  `4`   | `u32`        | `checksum` |
|  `8`   |`byte[size-8]`| `data`     |

To calculate the checksum, `checksum` is first set to zero, then, starting at `size`, the following algorithm is employed.


```c
struct nvram_buf_s {
  uint32_t size;
  uint32_t checksum;
  uint8_t data[N];
};

// ...
// read file into nvram_data
// ...

struct nvram_buf_s nvram_buf;
uint32_t checksum_reported = nvram_buf.checksum;

nvram_buf.checksum = 0;

// notice that it's not &nvram_buf.data
uint32_t checksum_expected = hcs32(&nvram_buf, nvram_buf.size);

```